### ISSUE TITLE
Cannot scale down service: "Did not find any nodes with sufficient free memory: ..."

### ISSUE THREAD

**Issue by colinmollenhour** [kontena/kontena#1883]
I have a "galera/node" service that has following config:

```
  stateful: yes
  scaling: 3
  strategy: ha
  deploy_opts:
    min_health: 0.6
    wait_for_port: 8082
  dns: node.galera.primary.kontena.local
  affinity: 
    - label==db-node
```

I wanted to "drain" a node and remove it from the grid "nicely" so I removed the labels and ran:

```
$ kontena service scale galera/node 2
 [fail] Scaling galera/node to 2 instances      
 [error] Cannot find applicable node for service instance primary/galera/node-2: Did not find any nodes with sufficient free memory: 4526965760.0
```

The node that I was trying to drain was actually running the `galera.node-3` container so it probably isn't related to the removal of the labels so much as it is the "scale" operation just fails when the service is already running and using all available memory.

Sorry if this is a duplicate, I searched and couldn't find it if it is..

**Comment by kke** [comment]
I wonder if this is related to #1837 ?

**Comment by SpComb** [comment]
Sounds like #1823

That change will be in 1.1.2

**Comment by colinmollenhour** [comment]
Yes, looks like this will be fixed in 1.1.2 then. Thanks!
